client/extensions/woocommerce/woocommerce-services/state/shipping-label/selectors.js

```diff
@@ -152,10 +152,19 @@ const getPackagesErrors = ( values ) => mapValues( values, ( pckg ) => {
 		errors.box_id = translate( 'Please select a package' );
 	}
 
-	if ( ! pckg.weight || 'number' !== typeof pckg.weight || 0 > pckg.weight ) {
+	if ( isNaN( pckg.weight ) || ! isNumber( pckg.weight ) || 0 >= pckg.weight ) {
 		errors.weight = translate( 'Invalid weight' );
 	}
 
+	const hasInvalidDimensions = [
+		pckg.length,
+		pckg.width,
+		pckg.height,
+	].reduce( ( result, dimension ) => ( result || isNaN( dimension ) || ! isNumber( dimension ) || 0 >= dimension ), false );
```

Comment:
This is the same calculation as above. Maybe you could extract it to separate method like `isDimensionvalid`. 
with that method instead of `reduce` you could use `lodash.some` for much shorter code.